### PR TITLE
Remove cirq.google references from circuit_test and circuit_dag_test

### PR DESCRIPTION
**Pull request by balopat**
Remove cirq.google references from circuit_test and circuit_dag_test.

Related to #3737.

### CODE REVIEW

File: cirq/circuits/circuit_test.py [hunk 1]
```diff
@@ -581,44 +579,126 @@ def test_concatenate_with_device():
     assert len(cone) == 0
 
 
+class FakeXmonDevice(cirq.Device):
+    """A Fake XMon device that was created to ensure certain Device validation features are
```
Comment (95-martin-orion) [inline cirq/circuits/circuit_test.py:583]: I think it's important to note (either here or elsewhere in this file) that what we're testing is *not* the specific behavior of Foxtail and Bristlecone, but rather a set of common behaviors for which these processors are reasonable examples.

File: cirq/circuits/circuit_test.py [hunk 2]
```diff
@@ -581,44 +579,126 @@ def test_concatenate_with_device():
...
+        ops.XPowGate,
+        ops.YPowGate,
+        ops.ZPowGate,
+    ),
+    qubits={
+        cirq.GridQubit(0, 0),
+        cirq.GridQubit(1, 0),
+        cirq.GridQubit(0, 1),
+        cirq.GridQubit(1, 1),
+        cirq.GridQubit(0, 6),
```
Comment (95-martin-orion) [inline cirq/circuits/circuit_test.py:648]: This may generate confusion - I'd recommend `cirq.GridQubit.rect(2,7)` instead.

File: cirq/circuits/circuit_test.py [hunk 3]
```diff
@@ -3375,25 +3455,25 @@ def test_insert_operations_errors():
...
+    c = cirq.Circuit(device=foxy)
 
     with pytest.raises(ValueError, match='Unsupported qubit type'):
         # Wrong type of qubit.
         c.append(cirq.Z(cirq.NamedQubit('q')))
     with pytest.raises(ValueError, match='Qubit not on device'):
         # A qubit that's not on the device.
         c[:] = [cirq.Moment([cirq.Z(cirq.GridQubit(-5, 100))])]
-    c.append(cirq.Z(cirq.GridQubit(0, 0)))
+    c.append(cirq.Z(cirq.GridQubit(0, 1)))
```
Comment (95-martin-orion) [inline cirq/circuits/circuit_test.py:3466]: Why this change?

File: cirq/circuits/circuit_test.py [hunk 4]
```diff
@@ -336,19 +334,19 @@ def test_repr(circuit_cls):
 ])"""
     )
 
-    c = circuit_cls(device=cg.Foxtail)
+    c = circuit_cls(device=foxy)
     cirq.testing.assert_equivalent_repr(c)
-    assert repr(c) == f'cirq.{circuit_cls.__name__}(device=cirq.google.Foxtail)'
+    assert repr(c) == f'cirq.{circuit_cls.__name__}(device=cirq.circuits.circuit_test.foxy)'
```
Comment (95-martin-orion) [inline cirq/circuits/circuit_test.py:339]: Can this just be `device=foxy` instead? Same below.

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: cirq/circuits/circuit_dag_test.py [hunk 5]
```diff
@@ -109,9 +109,14 @@ def test_from_circuit():
     assert sorted(circuit.all_qubits()) == sorted(dag.all_qubits())
 
 
+class TestDevice(cirq.Device):
+    def validate_operation(self, operation: cirq.Operation) -> None:
```
Comment (dstrain115) [inline cirq/circuits/circuit_dag_test.py:113]: Can't you use cirq.UNCONSTRAINED_DEVICE?

File: cirq/circuits/circuit_test.py (hunk 1 above)
Comment (dstrain115) [reply cirq/circuits/circuit_test.py:583]: Yes, I would call this something more specific, like CZOnlyTestDevice or something like that.

File: cirq/circuits/circuit_test.py [hunk 6]
```diff
@@ -581,44 +579,126 @@ def test_concatenate_with_device():
...
+        return not (
+            isinstance(operation.gate, cirq.CZPowGate)
+            and any(isinstance(op.gate, cirq.CZPowGate) for op in moment.operations)
+        )
+
+    def __repr__(self):
+        return self._repr
+
+
+foxy = FakeXmonDevice(
```
Comment (dstrain115) [inline cirq/circuits/circuit_test.py:635]: Should this be capitalized?

File: cirq/circuits/circuit_test.py [hunk 7]
```diff
@@ -581,44 +579,126 @@ def test_concatenate_with_device():
...
+        cirq.GridQubit(0, 0),
+        cirq.GridQubit(1, 0),
+        cirq.GridQubit(0, 1),
+        cirq.GridQubit(1, 1),
+        cirq.GridQubit(0, 6),
+    },
+    name='cirq.circuits.circuit_test.foxy',
+)
+
+bcone = FakeXmonDevice(
```
Comment (dstrain115) [inline cirq/circuits/circuit_test.py:653]: I would capitalize this.

File: cirq/circuits/circuit_test.py (hunk 4 above)
Comment (balopat) [reply cirq/circuits/circuit_test.py:339]: yes!

File: cirq/circuits/circuit_dag_test.py (hunk 5 above)
Comment (balopat) [reply cirq/circuits/circuit_dag_test.py:113]: Ah, nice, let me try that

File: cirq/circuits/circuit_test.py (hunk 1 above)
Comment (balopat) [reply cirq/circuits/circuit_test.py:583]: Right, okay, I will remove the reference to anything Foxtail / Bristlecone and rename the variable names as well. I had originally "ValidatingTestDevice" but that was too generic, maybe I'll go back to that.

File: cirq/circuits/circuit_test.py (hunk 3 above)
Comment (balopat) [reply cirq/circuits/circuit_test.py:3466]: Otherwise there was an overlapping moment exception. I added the logic now to the testclass in `can_add_operation_into_moment` to call `super()` like Foxtail does, so now this change can be reverted.

File: cirq/circuits/circuit_test.py (hunk 4 above)
Comment (balopat) [reply cirq/circuits/circuit_test.py:339]: Oh, actually, it cannot, as this is then used above in the assert_equivalent_repr - so the testclass have to generate repr that actually works.

File: cirq/circuits/circuit_test.py (hunk 2 above)
Comment (balopat) [reply cirq/circuits/circuit_test.py:648]: done.

**Comment by balopat** [comment]
@95-martin-orion @dstrain115 ptal

**Review by 95-martin-orion — APPROVED** [review]
